### ISSUE TITLE
Should collections count their allocated memory?

### ISSUE THREAD

**Issue by albertsgarde** [DKerp/get-size#3]
The following example prints `25`
```
use get_size::GetSize;

pub fn main() {
    let mut vec = Vec::with_capacity(3000);
    vec.push(2u8);
    println!("{}", vec.get_size());
}
```
24 for the stack size of the vector and 1 for the size of its heap contents.
But the allocation the vector is pointing to actually has a size of 3000 bytes.

My main purpose for using this crate is to figure out how much memory my program is using at runtime, in which case I would want the above code to print `3024` instead of `25`.
Is there a reason it doesn't?

**Comment by DKerp** [comment]
Hello @albertsgarde ,

The is no good reason for it.

I wrote this crate primarily for a caching library of mine, which I would use for objects where the additional space taking by the not used capacity is neglectable, so I went for using `.len()` instead of `.capacity()` in my implementation.

But the more I think about it now, the more I realize how stupid that is. As you pointed out, for most applications accounting the real memory usage is essential. Just think about a pool of already allocated, but currently empty buffers. The current implementation would account them as being empty, even trough they might in total allocate an enormous amount of memory.

I have changed the code so the `std` objects now also account for the not yet used, but already allocated capacity.

An update should be available in a few minutes.
